src/GitVersionCore/IExecuteCore.cs

```diff
@@ -0,0 +1,10 @@
+using GitVersion.OutputVariables;
+
+namespace GitVersion
+{
+    public interface IExecuteCore
+    {
+        VersionVariables ExecuteGitVersion(Arguments arguments);
```

Comment:
I don't think it's not obvious why this needs to be named `Execute` at all. What it does is compute some version variables. Perhaps `GitVersionComputer.ComputeVersionVariables()` would be a better name? Thoughts?

Reply:
That's exactly why I needed your feedback, as I'm diving in the code, I change the way we create the references by using DI and with your help, adjusting the code to make it more readable.

Reply:
I changed `ExecuteCore.ExecuteGitVersion` with `GitVersionExecutor.ComputeVersionVariables`